src/container/internal.tsx

```diff
@@ -65,24 +65,21 @@ export default function InternalContainer({
 
   /**
    * The visual refresh AppLayout component needs to know if a child component
-   * has a high constrast sticky header. This is to make sure the background element
+   * has a high contrast sticky header. This is to make sure the background element
    * stays in the same vertical position as the header content.
    */
-  useEffect(
-    function handleHasStickyBackground() {
-      const shouldUpdateStickyBackground = isRefresh && isSticky && variant === 'full-page';
+  useEffect(() => {
+    const shouldUpdateStickyBackground = isSticky && variant === 'full-page' && setHasStickyBackground;
```

Comment:
Instead of checking `isRefresh`, we can check if `setHasStickyBackground` was provided by the context (only exists in visual refresh). Same outcome, but more specific to the use-case